iceoryx_posh/include/iceoryx_posh/internal/popo/sample.inl

```diff
@@ -64,6 +64,19 @@ inline T* Sample<T>::operator->() noexcept
     return get();
 }
 
+template <typename T>
+template <typename S, typename>
+inline S& Sample<T>::operator*() noexcept
```

Comment:
Should we also introduce const overloads for `operator*`, `operator->`? Since it makes sense for the publisher to call maybe some kind of method with a const sample reference argument?

Reply:
Can probably do this.